src/recordlinker/schemas/tuning.py

```diff
@@ -90,3 +93,36 @@ def from_tuning_job(cls, job: TuningJob, request: fastapi.Request) -> typing.Sel
         """
         url: str = str(request.url_for("get-tuning-job", job_id=job.id))
         return cls(**job.model_dump(), status_url=pydantic.HttpUrl(url))
+
+
+@dataclasses.dataclass
+class TuningPair:
+    """
+    A pair of PIIRecords that are used for training a model.
+    """
+    record1: PIIRecord
+    record2: PIIRecord
+    sample_used: typing.Optional[int] = None  # the number of records sampled from to produce the pair
+
+    @classmethod
+    def from_data(cls, record1: dict, record2: dict, sample_used: int | None = None) -> typing.Self:
+        """
+        Contruct a TuningPair from raw PII data dictionaries.
+        """
+        return cls(
+            record1=PIIRecord.from_data(record1),
+            record2=PIIRecord.from_data(record2),
+            sample_used=sample_used,
+        )
+
+
+@dataclasses.dataclass
+class TuningProbabilities:
+    """
+    A dict of feature specific probabilities that two records are likely to belong to
+    the same cluster given a collection of TuningPairs. Also included are the number,
+    count, of TuningPairs analyzed and an optional parameter for sample size used.
+    """
+    probs: dict[Feature, float]  # the feature specific probabilities
+    count: int  # the number of pairs analyzed
+    sample_used: typing.Optional[int] = None  # the number of records sampled from to produce the probabilities
```

Comment:
What's the thinking on including these as variables in the data class rather than just tracking them in the function in which they're used? For example, in `calculate_class_probs` in prob_calc, we could start a variable `count = 0` outside the for loop over the iterator, increase it by 1 each iteration, and then divide the dictionary values here by `count` at the bottom of the function without actually storing count in the loop. Any reason to preserve it?